### DISCUSSION TITLE
out of range for slice of length

### DISCUSSION THREAD

**Discussion by j2eekingcn** [filecoin-project/lotus#7950] [Storage Provider]
miner window post failed

out of range for slice of length

2021-03-05T05:46:31.301 DEBUG merkletree::merkle > generated partial_tree of row_count 4 and len 585 with 8 branches for proof at 121800669
2021-03-05T05:46:31.301 DEBUG merkletree::merkle > generated partial_tree of row_count 4 and len 585 with 8 branches for proof at 108495240
2021-03-05T05:46:31.325 INFO storage_proofs_core::compound_proof > vanilla_proofs:finish
2021-03-05T05:46:31.731 INFO storage_proofs_core::compound_proof > snark_proof:start
2021-03-05T05:46:31.856 INFO bellperson::groth16::prover > Bellperson 0.12.5 is being used!
2021-03-05T05:46:56.860 INFO bellperson::groth16::prover > starting proof timer
2021-03-05T05:46:59.156 DEBUG bellperson::gpu::locks > Acquiring priority lock at "/data/tmpdir/bellman.priority.lock" ...
2021-03-05T05:46:59.156 DEBUG bellperson::gpu::locks > Priority lock acquired!
2021-03-05T05:46:59.372 INFO bellperson::gpu::locks > GPU is available for FFT!
2021-03-05T05:46:59.372 DEBUG bellperson::gpu::locks > Acquiring GPU lock at "/data/tmpdir/bellman.gpu.lock" ...
2021-03-05T05:46:59.372 DEBUG bellperson::gpu::locks > GPU lock acquired!
2021-03-05T05:46:59.469 INFO bellperson::gpu::fft > FFT: 1 working device(s) selected.
2021-03-05T05:46:59.470 INFO bellperson::gpu::fft > FFT: Device 0: GeForce RTX 2080 Ti
2021-03-05T05:46:59.470 INFO bellperson::domain > GPU FFT kernel instantiated!
2021-03-05T05:47:24.467 DEBUG bellperson::gpu::locks > GPU lock released!
thread '<unnamed>' panicked at 'range end index 6043200150033888668 out of range for slice of length 60780374232', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bellperson-0.12.5/src/groth16/mapped_params.rs:141:16
stack backtrace:
   0:          0x2576ca0 - std::backtrace_rs::backtrace::libunwind::trace::he85dfb3ae4206056
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:          0x2576ca0 - std::backtrace_rs::backtrace::trace_unsynchronized::h1ad28094d7b00c21
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:          0x2576ca0 - std::sys_common::backtrace::_print_fmt::h901b54610713cd21
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/sys_common/backtrace.rs:79
   3:          0x2576ca0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hb0ad78ee1571f7e0
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/sys_common/backtrace.rs:58
   4:          0x25e545c - core::fmt::write::h1857a60b204f1b6a
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/core/src/fmt/mod.rs:1080
   5:          0x2568952 - std::io::Write::write_fmt::hf7b7d7b243f84a36
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/io/mod.rs:1516
   6:          0x257ba7d - std::sys_common::backtrace::_print::hd093978a5287b8ff
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/sys_common/backtrace.rs:61
   7:          0x257ba7d - std::sys_common::backtrace::print::h20f46787581d56d7
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/sys_common/backtrace.rs:48
   8:          0x257ba7d - std::panicking::default_hook::{{closure}}::h486cbb4b82ffc357
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/panicking.rs:208
   9:          0x257b728 - std::panicking::default_hook::h4190c9e3edd4d591
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/panicking.rs:227
  10:          0x257c2a1 - std::panicking::rust_panic_with_hook::h72e78719cdda225c
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/panicking.rs:577
  11:          0x257be49 - std::panicking::begin_panic_handler::{{closure}}::h8bd07dbd34150a96
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/panicking.rs:484
  12:          0x257712c - std::sys_common::backtrace::__rust_end_short_backtrace::hdb6b3066ad29028a
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/sys_common/backtrace.rs:153
  13:          0x257be09 - rust_begin_unwind
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/std/src/panicking.rs:483
  14:          0x25e1781 - core::panicking::panic_fmt::hb15d6f55e8472f62
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/core/src/panicking.rs:85
  15:          0x25e7612 - core::slice::index::slice_end_index_len_fail::hd6713db859210b4a
                               at /rustc/beb5ae474d2835962ebdf7416bd1c9ad864fe101/library/core/src/slice/index.rs:41
  16:          0x1f8c204 - bellperson::groth16::mapped_params::read_g1::h004f43cae7169a0b
  17:          0x1e8fa9f - <core::iter::adapters::ResultShunt<I,E> as core::iter::traits::iterator::Iterator>::next::h6c8c6ce98d19654b
  18:          0x2767520 - <alloc::vec::Vec<T> as alloc::vec::SpecFromIter<T,I>>::from_iter::h8a6371b1a6f7802d
  19:          0x1f99b5e - <&bellperson::groth16::mapped_params::MappedParameters<E> as bellperson::groth16::params::ParameterSource<E>>::get_h::h949f5f6ca34ae937
  20:          0x1e826a4 - <core::iter::adapters::ResultShunt<I,E> as core::iter::traits::iterator::Iterator>::next::h3634294668c630ae
  21:          0x2769f43 - <alloc::vec::Vec<T> as alloc::vec::SpecFromIter<T,I>>::from_iter::ha59406e4b13af32a
  22:          0x1ec321d - core::iter::adapters::process_results::he7aadfffdfa06262
  23:          0x203d7d3 - bellperson::groth16::prover::create_random_proof_batch_priority::h276c650f8150c79e
  24:          0x29246c4 - storage_proofs_core::compound_proof::CompoundProof::circuit_proofs::h77cfb9f28969c7e1
  25:          0x2926e9f - storage_proofs_core::compound_proof::CompoundProof::prove::ha227d03058c967dd
  26:          0x2111259 - filecoin_proofs::api::post::generate_window_post::h3ce375a36a3720e3
  27:          0x1eecb91 - filecoin_proofs_api::post::generate_window_post_inner::h2412de5044ab1125
  28:          0x1eebd70 - filecoin_proofs_api::post::generate_window_post::h98bd97b00231eae4
  29:          0x1c9bdf1 - <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::hec3921fd7d7125a0
  30:          0x260c9de - ffi_toolkit::catch_panic_response::he354153ed6d2e058
  31:          0x1ddf6f0 - fil_generate_window_post
  32:          0x1b50ae6 - _cgo_a8fa62747d41_Cfunc_fil_generate_window_post
                               at /tmp/go-build/cgo-gcc-prolog:596
  33:           0x5a7370 - runtime.asmcgocall
                               at /usr/local/go/src/runtime/asm_amd64.s:656
2021-03-05T05:47:25.427 DEBUG bellperson::gpu::locks > Priority lock released!
2021-03-05T05:47:25.616+0800	[34mINFO[0m	storageminer	storage/wdpost_run.go:595	computing window post	{"batch": 0, "elapsed": 55.337996454}
2021-03-05T05:47:25.616+0800	[31mERROR[0m	storageminer	storage/wdpost_run.go:104	runPost failed: running window post failed:
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
        /data/lotus_sources/lotus/lotus/storage/wdpost_run.go:630
  - Rust panic: no unwind information
    github.com/filecoin-project/filecoin-ffi.GenerateWindowPoSt
    	/data/lotus_sources/lotus/lotus/extern/filecoin-ffi/proofs.go:587
    github.com/filecoin-project/lotus/extern/sector-storage/ffiwrapper.(*Sealer).GenerateWindowPoSt
    	/data/lotus_sources/lotus/lotus/extern/sector-storage/ffiwrapper/verifier_cgo.go:45
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
    	/data/lotus_sources/lotus/lotus/storage/wdpost_run.go:592
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runGeneratePoST
    	/data/lotus_sources/lotus/lotus/storage/wdpost_run.go:102
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).startGeneratePoST.func1
    	/data/lotus_sources/lotus/lotus/storage/wdpost_run.go:86
    runtime.goexit
    	/usr/local/go/src/runtime/asm_amd64.s:1374
2021-03-05T05:47:25.616+0800	[31mERROR[0m	storageminer	storage/wdpost_run.go:48	Got err running window post failed:
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
        /data/lotus_sources/lotus/lotus/storage/wdpost_run.go:630
  - Rust panic: no unwind information
    github.com/filecoin-project/filecoin-ffi.GenerateWindowPoSt
    	/data/lotus_sources/lotus/lotus/extern/filecoin-ffi/proofs.go:587
    github.com/filecoin-project/lotus/extern/sector-storage/ffiwrapper.(*Sealer).GenerateWindowPoSt
    	/data/lotus_sources/lotus/lotus/extern/sector-storage/ffiwrapper/verifier_cgo.go:45
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
    	/data/lotus_sources/lotus/lotus/storage/wdpost_run.go:592
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runGeneratePoST
    	/data/lotus_sources/lotus/lotus/storage/wdpost_run.go:102
    github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).startGeneratePoST.func1
    	/data/lotus_sources/lotus/lotus/storage/wdpost_run.go:86
    runtime.goexit
    	/usr/local/go/src/runtime/asm_amd64.s:1374 - TODO handle errors
2021-03-05T05:47:25.616+0800	[33mWARN[0m	storageminer	storage/wdpost_changehandler.go:249	Aborted window post Proving (Deadline: &{CurrentEpoch:552933 PeriodStart:552903 Index:0 Open:552903 Close:552963 Challenge:552883 FaultCutoff:552833 WPoStPeriodDeadlines:48 WPoStProvingPeriod:2880 WPoStChallengeWindow:60 WPoStChallengeLookback:20 FaultDeclarationCutoff:70})
2021-03-05T05:47:30.211+0800	[33mWARN[0m	storageminer	storage/wdpost_run.go:226	Checked sectors	{"checked": 211, "good": 211}

**Comment by Reiers** [comment]
Hi @j2eekingcn 

Thanks for the report. The issue you are experiencing lies elsewhere.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing Lotus issue ticket.

**Comment by LexLuthr** [comment]
@j2eekingcn Do you still need help with this problem? If not, then can you please tell us how you fixed it?

**Reply by LexLuthr** [reply]
I am archiving this discussion as there has been no new update from the author.
